blackbox/docs/general/builtins/window-functions.rst

```diff
@@ -314,6 +314,51 @@ Example::
    +------+-----------------------------------------------+
    SELECT 4 rows in set (... sec)
 
+.. _window-function-lag:
+
+``lag(arg [, offset [, default] ])``
+------------------------------------
+
+.. note::
+
+   The ``lag`` window function is an :ref:`enterprise feature
+   <enterprise-features>`.
+
+Synopsis
+........
+
+::
+
+   lag(argument any [, offset integer [, default any]])
+
+Returns the argument value evaluated at the row that precedes the current row
+by the offset within the partition. If there is no such row, the return value
+is ``default``. If ``offset`` or ``default`` arguments are missing, they
+default to ``1`` and ``null``, respectively.
+
+Both ``offset`` and ``default`` are evaluated with respect to the current row.
+
+If ``offset`` is ``0``, then argument value is evaluated for the current row.
+
+The ``default`` and ``argument`` data types must match.
+
+Example::
+
+   cr> SELECT
+   ...   col1,
+   ...   LAG(col1) OVER(PARTITION BY col1 > 2) lag
```

Comment:
I think it would be easier to follow for users if we've a dedicated partition column so that we don't offload the `col1 > 2` computation to users reading/trying to understand this.